File: app/src/main/java/com/awscherb/cardkeeper/data/model/ScannedCode.java
package com.awscherb.cardkeeper.data.model;

import com.google.zxing.BarcodeFormat;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class ScannedCode extends RealmObject implements BaseModel {

    @PrimaryKey
    private long id;

    private String format;

    private String text;

    private String title;

    //================================================================================
    // Getters
    //================================================================================

    @Override
    public long getId() {
        return id;
    }

    public BarcodeFormat getFormat() {
        return BarcodeFormat.valueOf(format);
    }

    public String getText() {
        return text;
    }

    public String getTitle() {
        return title;
    }

    //================================================================================
    // Setters
    //================================================================================

    public void setId(long id) {
        this.id = id;
    }

    public void setFormat(BarcodeFormat format) {
        this.format = format.toString();
    }

    public void setText(String text) {
        this.text = text;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: app/src/main/java/com/awscherb/cardkeeper/data/model/ScannedCode.kt
package com.awscherb.cardkeeper.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.zxing.BarcodeFormat

@Entity(tableName = "scannedCode")
data class ScannedCode(
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0,
    var format: BarcodeFormat,
    var text: String,
    var title: String,
    var created: Long = 0
) : BaseModel()